SocialCareCaseViewerApi.Tests/V1/Data/ASCEmergencyIdImportTests.cs

```diff
@@ -0,0 +1,178 @@
+                   do
+                    $$
+                            declare
+                                    new_person record;
+                                    new_person_mosaic_id bigint;
+                            begin
+                                    for new_person
+                                    in SELECT
+                                            imports.first_name,
+                                            imports.last_name,
+                                            imports.full_name,
+                                            imports.date_of_birth,
+                                            imports.person_id,
+                                            imports.gender,
+                                            imports.restricted,
+                                            imports.context_flag
+
+                            FROM
+                                    dbo.SCCV_PERSONS_LOOKUP as lookup RIGHT OUTER JOIN
+                                    dbo.SCCV_PERSONS_IMPORT as imports ON lookup.NC_ID = imports.PERSON_ID
+                            WHERE lookup.PERSON_ID is null AND LOWER(imports.PERSON_ID) like 'tmp%'
+
+                            loop
+                                    new_person_mosaic_id = null;
+
+                                    insert into dbo.dm_persons(
+                                            first_name,
+                                            last_name,
+                                            full_name,
+                                            date_of_birth,
+                                            gender,
+                                            restricted,
+                                            context_flag,
+                                            from_dm_person
+                                    )
+                                    values (
+                                            new_person.first_name,
+                                            new_person.last_name,
+                                            new_person.full_name,
+                                            new_person.date_of_birth,
+                                            new_person.gender,
+                                            new_person.restricted,
+                                            new_person.context_flag,
+                                            'N'
+                                    )
+                                    returning person_id into new_person_mosaic_id;
+
+                                    insert into dbo.sccv_persons_lookup(person_id, nc_id)
+                                            values (new_person_mosaic_id, new_person.person_id);
+
+                                    raise notice '% - %', new_person.person_id, new_person_mosaic_id;
+                            end loop;
+                    end;
+                    $$
+                   ";
```

Comment:
Can we do this without large query strings that are executed as SQL?

I think I understand what the tests here are doing but it definitely makes the tests harder to understand.
If we consider tests as proof our code does what we say it does we want to make our tests as readable as possible.

Reply:
These tests are a bit odd since they test SQL scripts designed to be run manually. They are there just to give more confidence that they actually do what the tests say. Idea is that you should be able to copy and paste that script to SQL server and run it. We are slowly introducing more features in the app to cover these random import/update scripts and we really want to get away from them altogether. There might be better ways to set up the tests, but I couldn't think of any. Happy to review for future scripts to make them easier to manage

Reply:
Ok, that makes sense for now :+1: